KalturaWowzaServer/src/main/java/com/kaltura/media/server/wowza/CuePointsManager.java

```diff
@@ -86,114 +94,37 @@ public LiveStreamPacketizerDataHandler(String streamName) {
 			this.streamName = streamName;
 		}
 
-		private String jsonEscape(String string) {
-			if (string == null || string.length() == 0) {
-				return "";
-			}
-
-			char         c;
-			int          i;
-			int          len = string.length();
-			StringBuilder sb = new StringBuilder(len + 4);
-			String       t;
-
-			for (i = 0; i < len; i += 1) {
-				c = string.charAt(i);
-				switch (c) {
-					case '\\':
-					case '"':
-						sb.append('\\');
-						sb.append(c);
-						break;
-					case '/':
-						sb.append('\\');
-						sb.append(c);
-						break;
-					case '\b':
-						sb.append("\\b");
-						break;
-					case '\t':
-						sb.append("\\t");
-						break;
-					case '\n':
-						sb.append("\\n");
-						break;
-					case '\f':
-						sb.append("\\f");
-						break;
-					case '\r':
-						sb.append("\\r");
-						break;
-					default:
-						if (c < ' ') {
-							t = "000" + Integer.toHexString(c);
-							sb.append("\\u").append(t.substring(t.length() - 4));
-						} else {
-							sb.append(c);
-						}
-				}
+		private void addToMap(Map<String, Object> map, AMFDataObj o, String key) {
+			AMFData obj = o.get(key);
+			if (obj != null) {
+				map.put(key, obj.getValue());
+			} else {
+				logger.debug(key + " is not found in AMFDataObj");
 			}
-			return sb.toString();
 		}
 
-		private String jsonAMF(AMFData data) {
-			StringBuilder sb = new StringBuilder();
-			int t = data.getType();
+		private String jsonAMF(AMFDataObj dataObj) {
 
-			if (t == AMFData.DATA_TYPE_STRING || t == AMFData.DATA_TYPE_DATE) {
-				sb.append("\"").append(jsonEscape(data.toString())).append("\"");
+			if (dataObj == null) {
+				logger.debug("data object is null");
+				return null;
```

Comment:
Same here - should this ever happen? If not throw an exception instead of returning null